app/indexing.py

```diff
@@ -208,8 +210,29 @@ def sort_query_words(query_words):
     #       which are usually more effective filters, come first.
     return sorted(sorted_query_words, key=len, reverse=True)
 
-
-def search(repo, query_obj, max_results):
+def create_query(query):
```

Comment:
since it's only used in search_with_index, let's define it under the function.

def search_with_index(repo, query_obj, max_results):
  """ ... """
  def  create_query(query):
    something
  something